src/api_client.rs

```diff
@@ -119,9 +119,9 @@ impl RateLimitedClient {
     /// * `quota` - Outgoing request quota in requests per second
     pub fn new(nyt_s: &str, quota: NonZeroU32) -> Self {
```

Comment:
What do you think about supporting either the cookie or the header option to preserve the header-based authentication option? Maybe passing in something like:

```rust
enum Auth<'a> {
  NytSCookie(&'a str)
  NytSHeader(&'a str)
}
```

The cookie method seems like the most robust option now, but fwiw the header option still works for me with my old NYT-S token so I'd like to keep supporting it (even if we leave the header option undocumented in the README).

Reply:
And fwiw I think this is low-priority, so feel free to punt on this if you don't have the time/energy. Just trying to selfishly preserve my old workflow just in case the cookie-based token expires sooner than the header-based token, which I  haven't had to refresh in multiple years of running this script.

Reply:
How about this hacky, undocumented way of preserving your workflow: auto-detect whether it's `nyt-s` (header) or `NYT-S` (cookie) based on string length? If that works, can you let me know the string length of the `nyt-s` header?

Reply:
My `nyt-s` header is 142. I'm fine with the hack. It'll be easy to clean up later if needed.

Reply:
Cool, all changes are now in -- let me know how this looks.